### PR TITLE
TraceView: Refactor theming to use Grafana theme directly

### PR DESCRIPTION
**Pull request by aocenas**
Part of https://github.com/grafana/grafana/issues/43661

Removes all the bindings and contexts to propagate the values of the theme from TraceView deeper and just directly use Grafana theme from grafana/ui.

### CODE REVIEW

File: packages/jaeger-ui-components/src/TraceTimelineViewer/SpanBar.tsx [hunk 1]
```diff
@@ -17,14 +17,16 @@ import { css } from '@emotion/css';
...
+import { stylesFactory } from '@grafana/ui';
+import { autoColor } from '../Theme';
 import { TraceSpan } from '../types/trace';
 import { TNil } from '../types';
 import { UIPopover } from '../uiElementsContext';
 import AccordianLogs from './SpanDetail/AccordianLogs';
 import { ViewedBoundsFunctionType } from './utils';
 
-const getStyles = createStyle((theme: Theme) => {
+const getStyles = stylesFactory((theme: GrafanaTheme2) => {
```
Comment (ivanahuckova) [inline packages/jaeger-ui-components/src/TraceTimelineViewer/SpanBar.tsx:29]: SpanBar is functional component so we can use `useStyles2` and remove theme passed from `SpanBarRow`?

**Review by ivanahuckova — COMMENTED** [review]
Really nice work with this 👍 . I've left 1 comment.

**Review by connorlindsey — CHANGES_REQUESTED** [review]
I think as part of this refactor, we should use the grafana theme colors. In the same way `safeSize` was removed, we should be able to remove `autoColor` and any hard-coded color values.

**Comment by aocenas** [comment]
@connorlindsey yeah definitely want to do that just want to keep the PRs small so would rather do that separately. Will add that to the https://github.com/grafana/grafana/issues/43661 to track it.

**Comment by connorlindsey** [comment]
Sounds good to me!

**Review by connorlindsey — APPROVED** [review]

File: packages/jaeger-ui-components/src/TraceTimelineViewer/SpanBar.tsx (hunk 1 above)
Comment (aocenas) [reply packages/jaeger-ui-components/src/TraceTimelineViewer/SpanBar.tsx]: good point, changed

Comment (ivanahuckova) [reply packages/jaeger-ui-components/src/TraceTimelineViewer/SpanBar.tsx]: LGTM now!

**Review by ivanahuckova — APPROVED** [review]
👍

**Pull request merged**